src/common/ceph_time.h

```diff
@@ -468,6 +468,20 @@ inline timespan to_timespan(signedspan z) {
   ceph_assert(z >= signedspan::zero());
   return std::chrono::duration_cast<timespan>(z);
 }
+
+// detects presence of Clock::to_timeval() and from_timeval()
+template <typename Clock, typename = std::void_t<>>
+struct converts_to_timespec : std::false_type {};
+
+template <typename Clock>
+struct converts_to_timespec<Clock, std::void_t<decltype(
+    Clock::from_timespec(Clock::to_timespec(
+        std::declval<typename Clock::time_point>()))
+  )>> : std::true_type {};
+
```

Comment:
I am pleased with this update! And the void_t trick is much cleaner than the one I usually use. I'll have to start adopting it.

Reply:
yeah, it's neat but the name void_t doesn't really explain what it's doing. http://en.cppreference.com/w/cpp/experimental/is_detected will be an improvement